### PR TITLE
Extraction of Globals.prefs.put and .get

### PR DESCRIPTION
**Pull request by calixtus**
This PR aims to extract the calls to Globals.prefs.put and .get out of the JabRef codebase in preparation for some architectural healing...

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I added some further extractions and removed some calls to JabRefPreferences.getInstance(). TODO: We need some kind of MrDlib Preferences for SEND_LANGUAGE_DATA etc.

File: src/main/java/org/jabref/logic/crawler/StudyRepository.java [hunk 1]
```diff
@@ -280,7 +279,7 @@ private String findQueryIDByQueryString(String query) {
      * @param crawlResults The results that shall be persisted.
      */
     private void persistResults(List<QueryResult> crawlResults) throws IOException {
-        DatabaseMerger merger = new DatabaseMerger(keywordDelimiter);
+        DatabaseMerger merger = new DatabaseMerger(importFormatPreferences.getKeywordSeparator());
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/crawler/StudyRepository.java:282]: The keyword separator and delimiter are different the delimiter is groups related

Comment (calixtus) [reply src/main/java/org/jabref/logic/crawler/StudyRepository.java:282]: Oh ok, I see. Think I was a little bit too sloppy here...

Comment (calixtus) [reply src/main/java/org/jabref/logic/crawler/StudyRepository.java:282]: Ok, I checked it, it's a bit confusing, but I think, this is correct here:
`importFormatPreferences.getKeywordSeparator()` returns `prefs.getKeywordDelimiter()`, which is by default `,`. In this case, `keywordDelimiter` was originally initialized with an argument in the constructor in `StudyRepository.java`, which is called in `StartLiteratureReviewAction.java`, which is called by `JabRefFrame.java` with the argument `prefs.getKeywordDelimiter()`. As `prefs.getImportExportPreferences().getKeywordSeparator()` returns the same as `prefs.getKeywordDelimiter(), this should be right here: can you please doublecheck this?

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/crawler/StudyRepository.java:287]: Ah yeah, it's indeed the same. I overlooked that it's the same as in the ImportFormatPreferences, I thought it were two different ones.

**Comment by Siedlerchr** [comment]
We should wait with merging until #6187

**Comment by calixtus** [comment]
This is definitely enough...
Poor @tobiasdiez having to review everything. :wink:
But seriously, this is mostly refactoring, rewording, moving around etc.

Next step will be to extract all the calls to `Globals.prefs`.
I'm still thinking on how to deal with the preference migrations.

**Comment by calixtus** [comment]
The localization consistency tests are parsing comments too...

**Comment by Siedlerchr** [comment]
Can you test #7147  as well? Seems like also it's related to the prefs somehow

File: src/main/java/org/jabref/gui/ClipBoardManager.java [hunk 2]
```diff
@@ -47,15 +48,18 @@
...
+    public ClipBoardManager(PreferencesService preferencesService) {
+        this(Clipboard.getSystemClipboard(), Toolkit.getDefaultToolkit().getSystemSelection(), Globals.IMPORT_FORMAT_READER, preferencesService);
     }
 
-    public ClipBoardManager(Clipboard clipboard, java.awt.datatransfer.Clipboard primary, ImportFormatReader importFormatReader) {
+    public ClipBoardManager(Clipboard clipboard, java.awt.datatransfer.Clipboard primary, ImportFormatReader importFormatReader, PreferencesService preferencesService) {
         ClipBoardManager.clipboard = clipboard;
         ClipBoardManager.primary = primary;
         ClipBoardManager.importFormatReader = importFormatReader;
+
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/ClipBoardManager.java:61]: I guees Tobi or Olli will complain about the empty space ;)
```suggestion
```

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/ClipBoardManager.java:61]: No here it's fine...just don't add empty lines at the beginning or end of a method

**Review by tobiasdiez — APPROVED** [review]
This looks very nice! Thanks for all the hard work you put into the preferences. Let's merge to reduce the chance that incoming PRs create merge conflicts.

**Comment by calixtus** [comment]
Something's off with the tests. I'll look into it

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 3]
```diff
@@ -817,7 +817,7 @@ private MenuBar createMenu() {
                 new SeparatorMenuItem(),
 
                 factory.createMenuItem(StandardActions.SEND_AS_EMAIL, new SendAsEMailAction(dialogService, prefs, stateManager)),
-                pushToApplicationMenuItem,
+                pushToApplicationMenuItem
                 // Disabled until PR #7126 can be merged
                 // new SeparatorMenuItem(),
                 //factory.createMenuItem(StandardActions.START_SYSTEMATIC_LITERATURE_REVIEW,
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/JabRefFrame.java:823]:
```suggestion
                // factory.createMenuItem(StandardActions.START_SYSTEMATIC_LITERATURE_REVIEW,
```

**Pull request merged**